actionview/lib/action_view/helpers/asset_tag_helper.rb

```diff
@@ -77,12 +80,20 @@ module AssetTagHelper
       def javascript_include_tag(*sources)
         options = sources.extract_options!.stringify_keys
         path_options = options.extract!("protocol", "extname", "host", "skip_pipeline").symbolize_keys
-        sources.uniq.map { |source|
+        early_hints_links = []
+
+        sources_tags = sources.uniq.map { |source|
+          href = path_to_javascript(source, path_options)
+          early_hints_links << { link: href, as: "script" }
           tag_options = {
-            "src" => path_to_javascript(source, path_options)
+            "src" => href
           }.merge!(options)
           content_tag("script".freeze, "", tag_options)
```

Comment:
Would it make sense to change this to `tag.script(src: href, **options)` while we're here? (Though that would probably necessitate shifting to `symbolize_keys` above).

Reply:
Is that desirable in general in internal stuff? The improvement for user code is nice, but it feels like fairly pointless (and comparatively allocation-heavy) indirection for us.

Reply:
Right, it's a bit of a back and forth. I was focusing on removing the `freeze` call. Let's defer to some other time, since it's out of scope for this particular thing 👍